### ISSUE TITLE
Correct hashing of unitful values

### ISSUE THREAD

**Issue by bkamins** [PainterQubits/Unitful.jl#384]
Reposted from https://github.com/JuliaData/DataFrames.jl/issues/2486.

Currently we have:
```
julia> isequal(1u"d", 24u"hr")
true

julia> hash(1u"d"), hash(24u"hr")
(0xc0114c5d5037b321, 0x34c298a544e80792)
```
and `hash` requires that if two values pass `isequal` test they must have the same hash.

CC @tomyun

**Comment by giordano** [comment]
Related to #379 probably?

**Comment by bkamins** [comment]
Yes - sorry for not knowing this package well enough. Buy my recommendation is - either make the values fail `isequal` test or align their `hash` values. The invariant:

> Compute an integer hash code such that `isequal(x,y)` implies  `hash(x)==hash(y)`

is fundamental for composability.
